src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -207,10 +233,17 @@ private TrieNode searchForJsonPathKeyNode(byte[] bytes, Iterator<MaskingState.Se
             if (node == null) {
                 return null;
             }
-            MaskingState.SegmentReference segmentReference = jsonPath.next();
-            int keyStartIndex = segmentReference.start;
-            int keyLength = segmentReference.offset;
-            for (int i = keyStartIndex; i < keyStartIndex + keyLength; i++) {
+            JsonPathSegmentReference jsonPathSegmentReference = jsonPath.next();
+            int keyOffset = jsonPathSegmentReference.getOffset();
+            int keyStart = jsonPathSegmentReference.getLength();
+            if (jsonPathSegmentReference.isArraySegment()) {
+                node = node.children[Character.forDigit(keyOffset, DECIMAL_RADIX) + BYTE_OFFSET];
```

Comment:
I just realised that it won't work when the array index is > 10. I will push a fix and tests that cover such use cases.

Reply:
Strange that we didn't catch that in our fuzzing tests

Reply:
Ah 😅 
<img width="397" alt="image" src="https://github.com/Breus/json-masker/assets/15277543/3c0a709f-cc3e-4fe0-a5d2-34a091fca87e">

@Breus I think we should use at least size 10 for all generation options

Reply:
Lol

Reply:
Maybe also increase some other values to be sure?

Reply:
Yes, I think we need to do it for all of them, so it's at least 10

Reply:
@donavdey can you update them in your PR?

Reply:
@Breus I've tried increasing all of them to 12, but none of the tests failed 🤷  Apparently Java decided that it's better to return a null character `\u0000` if input is incorrect and then our trie simply doesn't have null characters to match. We'll need a specific test to cover that

Reply:
With this
```java
                char digit = Character.forDigit(keyOffset, DECIMAL_RADIX);
                if (digit == '\u0000') {
                    throw new IllegalStateException("Invalid digit " + keyOffset);
                }
                node = node.children[digit + BYTE_OFFSET];
```
we actually have fuzzing test failure

Reply:
Added a check for null character